### PR TITLE
migrate winlogbeat pipeline

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

## How does this PR solve the problem?

Migrates the `winlogbeat` pipeline

Example of the pipeline:
https://buildkite.com/elastic/beats-winlogbeat/builds/356

Examples of dependencies:
https://buildkite.com/elastic/beats-libbeat/builds/1192
https://buildkite.com/elastic/beats-metricbeat/builds/2405
https://buildkite.com/elastic/beats-packetbeat/builds/1187

## Related issues

https://github.com/elastic/ingest-dev/issues/1693

### CODE REVIEW

File: .buildkite/scripts/common.sh [hunk 1]
```diff
@@ -29,6 +29,10 @@ packetbeat_changeset=(
   "^packetbeat/.*"
   )
 
+winlogbeat_changeset=(
+  "^metricbeat/.*"
```
Comment (v1v) [inline .buildkite/scripts/common.sh:33]: typo?

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+    if: "build.source == 'ui'"
+    allow_dependency_failure: false
+
+  - label: ":linux: Load dynamic winlogbeat pipeline"
+    key: "winlogbeat-pipeline"
+    command: ".buildkite/scripts/generate_winlogbeat_pipeline.sh"
+    agents:
+      provider: "gcp"
+      image: "${IMAGE_UBUNTU_X86_64}"
+      machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
```
Comment (v1v) [inline .buildkite/winlogbeat/pipeline.winlogbeat.yml:39]: The default k8s agent is enough to run `buildkite-agent pipeline upload $pipelineName`

**Review by v1v — CHANGES_REQUESTED** [review]

File: .buildkite/winlogbeat/pipeline.winlogbeat.yml (hunk 2 above)
Comment (sharbuz) [reply .buildkite/winlogbeat/pipeline.winlogbeat.yml]: fixed

File: .buildkite/scripts/common.sh (hunk 1 above)
Comment (sharbuz) [reply .buildkite/scripts/common.sh]: fixed

**Review by v1v — APPROVED** [review]

**Review by dliappis — APPROVED** [review]
LGTM thanks!

**Pull request merged**